Integrations/src/test/java/io/deephaven/integrations/learn/IndexSetTest.java

```diff
@@ -0,0 +1,81 @@
+package io.deephaven.integrations.learn;
+
+import junit.framework.TestCase;
+import org.junit.Test;
+
+import java.util.NoSuchElementException;
+import java.util.PrimitiveIterator;
+
+public class IndexSetTest {
+
+    static IndexSet indexSet;
+
+    @Test
+    public void AddVerifyRemoveTest() throws Exception {
+        indexSet = new IndexSet(5);
+
+        TestCase.assertEquals(-1, indexSet.current);
+        TestCase.assertEquals(5, indexSet.maxSize);
+
+        indexSet.add(135);
+        indexSet.add(54);
+        indexSet.add(100000);
+
+        PrimitiveIterator.OfLong itr = indexSet.iterator();
+        TestCase.assertEquals(3, indexSet.size());
+        TestCase.assertEquals(135, (long) itr.next());
+        TestCase.assertEquals(54, (long) itr.next());
+        TestCase.assertEquals(100000, (long) itr.next());
+
+        indexSet.clear();
+
+        TestCase.assertEquals(-1, indexSet.current);
+        TestCase.assertEquals(5, indexSet.maxSize);
+    }
+
+    @Test(expected = Exception.class)
+    public void TooManyElementsTest() throws Exception {
+        indexSet = new IndexSet(5);
+
+        for (int i = 0 ; i < 6 ; i++) {
+            indexSet.add(i);
+        }
+        indexSet.clear();
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void ParseTooFarTest() throws Exception {
+        indexSet = new IndexSet(5);
+
+        indexSet.add(135);
+        indexSet.add(54);
+        indexSet.add(100000);
+
+        PrimitiveIterator.OfLong itr = indexSet.iterator();
+        for (int i = 0 ; i < 5 ; i++) {
+            itr.next();
+        }
+        indexSet.clear();
```

Comment:
Your test is too loose.  It adds 3 values but iterates for 5.  The implementation could be incorrect and still pass the test.  Check the other tests for this problem as well.

Reply:
I don't think I understand this. This test is annotated with `expected = NoSuchElementException` so that when it goes too far, it throws the correct error and passes the test. I'm saying that maximum size of the IndexSet is 5, but only adding 3 values, so it should not be able to iterate for 5 without throwing this error. Are you saying it should be 4?